validator/postprocess.py

```diff
@@ -29,26 +32,81 @@ def step_into(self, datafile_schema: str):
             self.schema_path.pop()
 
 
-def resolve_resource_references(bundle: Bundle):
+def itemgetter(properties, obj):
+    def to_hashable(field):
+        if isinstance(field, collections.Hashable):
+            return field
+        else:
+            return repr(field)
+    if len(properties) == 1:
+        item = properties[0]
+        return to_hashable(obj.get(item))
+    else:
+        id = tuple(to_hashable(obj.get(item)) for item in properties)
+        if set(id) == {None}:
+            return None
+        else:
+            return id
+
+
+def ensure_context_uniqueness(df, df_path: str, config: list):
```

Comment:
- Since this function uses recursion, there are chances we could run into call stack limit. Could we perhaps use a local stack?
- Is there a case where we might observe endless recursion/loop?

Reply:
this function operators on a single datafile `df` and does not follow references. therefore we are safe from endless recursion. 

following refs is not necessary to calculate identifiers because a ref is represented in a datafile as a single `$ref` attribute, which provides enough information for uniqueness within the context of a list.